### DISCUSSION TITLE
https for github is unavailable most of the time, can ssh be supported?

### DISCUSSION THREAD

**Discussion by playgithub** [microsoft/vcpkg#19090] [Ideas]
**Problem**
In my country, `git clone` from github in `https` fails most of the time because of dns problems, it leads to `vcpkg install` failures.

**Proposed solution**
support for `ssh` for github, e.g. `git clone git@github.com:some-lib.git` works.

**Comment by NancyLi1013** [comment]
Hi @strega-nil-ms, @ras0219-msft 

Do we have a plan to support this?

**Comment by ras0219-msft** [comment]
This is really unfortunate, because falling back to `git` prevents us from using our SHA512 hash validation that you're getting the correct source files. `git` still generally uses SHA1 which has been shown to be vulnerable, which is why we avoid it as much as possible.

When you say DNS problems, could you go into more detail specifically about how `https://` fails? For example, how are you handling all the other cases where we can't use git, like
1. https://github.com/ninja-build/ninja/releases/download/v1.10.2/ninja-win.zip
2. https://github.com/Kitware/CMake/releases/download/v3.20.2/cmake-3.20.2-windows-i386.zip
3. https://github.com/git-for-windows/git/releases/download/v2.26.2.windows.1/PortableGit-2.26.2-32-bit.7z.exe

**Reply by playgithub** [reply]
> When you say DNS problems, could you go into more detail specifically about how `https://` fails? For example, how are you handling all the other cases where we can't use git, like
> 
> 1. https://github.com/ninja-build/ninja/releases/download/v1.10.2/ninja-win.zip
> 2. https://github.com/Kitware/CMake/releases/download/v3.20.2/cmake-3.20.2-windows-i386.zip
> 3. https://github.com/git-for-windows/git/releases/download/v2.26.2.windows.1/PortableGit-2.26.2-32-bit.7z.exe

It's DNS polution, the upstream hijacks DSN queries, the urls in the example fails most of the time.
Only a proxy can solve the problem.

**Comment by playgithub** [comment]
Solved by `git config --global url.ssh://git@github.com/.insteadOf https://github.com/`,
it will convert `https` to `ssh` automatically.
